src/main.rs

```diff
@@ -38,12 +70,37 @@ async fn start_xd() {
 
 /// Function to start a new instance of the kaenguru bot
 async fn start_kg() {
-    panic::set_hook(Box::new(|_| {
-        eprintln!("Fatal: Not discord token Känguru found!\nFatal: Please set the DISCORD_TOKEN_KAENGURU environment variable to your discord token!\nFatal: More information can be found here: https://mafelp.github.io/MCDC/installation#get-a-discord-bot-token");
-        exit(2);
-    }));
-    let kg_token = env::var("DISCORD_TOKEN_KAENGURU").expect("kg_token");
-    let _ = panic::take_hook();
+    // Get the token from the configuration
+    let kg_token = match env::var("DISCORD_TOKEN_KAENGURU") {
+        Ok(s) => s,
+        Err(_) => {
+            let config_arc = Arc::clone(&CONFIG);
+            let mut config_lock = config_arc.lock();
+            while config_lock.is_err() {
```

Comment:
A error in locking means the lock has been poisoned and can't be unlocked again.
This should panic instead of retrying!